matchzoo/zmain.py

```diff
@@ -0,0 +1,510 @@
+    for mobj in config['metrics']:
+        mobj = mobj.lower()
+        if '@' in mobj:
+            mt_key, mt_val = mobj.split('@', 1)
+            eval_metrics[mobj] = metrics.get(mt_key)(int(mt_val))
+        else:
+            eval_metrics[mobj] = metrics.get(mobj)
+
+    zmodel.compile(optimizer='adam', loss=zloss(batch=10))
+    print('[Model] Model Compile Done.', end='\n')
+    zoo_input_data = []
+    zoo_label = []
+    count = 0
+
+    for i_e in range(num_iters):
+        for tag, generator in train_gen.items():
+            genfun = generator.get_batch_generator()
+            for input_data, y_true_value in genfun:
+                count += 1
+                if count > 10:
+                    break
+                names = ['query', 'doc']
+                shapes = [(None, 10), (None, 40)]
+                list_input_data = standardize_input_data(input_data, names, shapes, check_batch_axis=False)
+                zoo_input_data.append(list_input_data)
+                y_true_value = np.expand_dims(y_true_value, 1)
+                zoo_label.append(y_true_value)
+
+    zinput, zlabel = gen_zinput(zoo_input_data, zoo_label)
+
+    zmodel.fit(zinput, zlabel, batch_size=280, nb_epoch=1, distributed=False)
+
+    for i_e in range(num_iters):
+        for tag, generator in eval_gen.items():
+            genfun = generator.get_batch_generator()
+            print('[%s]\t[Eval:%s] ' % (time.strftime('%m-%d-%Y %H:%M:%S', time.localtime(time.time())), tag), end='')
+            res = dict([[k,0.] for k in eval_metrics.keys()])
+            num_valid = 0
+            for input_data, y_true in genfun:
+                names = ['query', 'doc']
+                shapes = [(None, 10), (None, 40)]
+                list_input_data = standardize_input_data(input_data, names, shapes, check_batch_axis=False)
+                y_pred = zmodel.forward(list_input_data)
+                if issubclass(type(generator), inputs.list_generator.ListBasicGenerator):
+                    list_counts = input_data['list_counts']
+                    for k, eval_func in eval_metrics.items():
+                        for lc_idx in range(len(list_counts)-1):
+                            pre = list_counts[lc_idx]
+                            suf = list_counts[lc_idx+1]
+                            res[k] += eval_func(y_true = y_true[pre:suf], y_pred = y_pred[pre:suf])
+                    num_valid += len(list_counts) - 1
+                else:
+                    for k, eval_func in eval_metrics.items():
+                        res[k] += eval_func(y_true = y_true, y_pred = y_pred)
+                    num_valid += 1
+            generator.reset()
+            print('Iter:%d\t%s' % (i_e, '\t'.join(['%s=%f'%(k,v/num_valid) for k, v in res.items()])), end='\n')
+            sys.stdout.flush()
+        # if (i_e+1) % save_weights_iters == 0:
+        #     zmodel.save_weights(weights_file % (i_e+1))
+
+
+def standardize_input_data(data, names, shapes=None,
```

Comment:
try import this method like: `from keras.engine.training import _standardize_input_data`